hwmv2: port STM32L5 SoCs and boards

erwango opened the pull request.

erwango committed:
soc: st: Migrate stm32l5 series to new hw model

Migrate STM2L5 series to new HW model.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
boards: Convert nucleo_l552ze_q to HWM v2

Port the board to HWMv2.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
boards: Convert stm32l562e_dk to HWM v2

Port the board to HWMv2.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
soc: stm32l5: Update stm32l5 non secure targets in various places

Need to update board name scheme for L5 ns targets.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango commented on boards/st/nucleo_l552ze_q/Kconfig.nucleo_l552ze_q:
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2020 Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_NUCLEO_L552ZE_Q
+	select SOC_STM32L552XX
+
+config BOARD_NUCLEO_L552ZE_Q_STM32L552XX_NS
+	bool
+	select SOC_STM32L552XX
```
This seems shaky as when building ns target `west build -b  nucleo_l552ze_q/stm32l552xx/ns`, both symbols ends up being defined:
```
$ grep -r BOARD_NUCLEO_L552ZE_Q build/nucleo_l552ze_q/stm32l552xx/ns/zephyr/.config 
CONFIG_BOARD_NUCLEO_L552ZE_Q=y
CONFIG_BOARD_NUCLEO_L552ZE_Q_STM32L552XX_NS=y

```

erwango replied:
@tejlmand, @nordicjm. I would welcome advices

thedjnK replied:
This is expected, `BOARD_NUCLEO_L552ZE_Q` is the board itself and `BOARD_NUCLEO_L552ZE_Q_STM32L552XX_NS` is the variant

erwango replied:
ok, I'll keep it that way then. Txs

erwango committed:
soc: stm32: Set default on USE_DT_CODE_PARTITION

Rather than configuring USE_DT_CODE_PARTITION in all ns targets,
do it conditionally as part of SoC description.

This allows cleaning up some BOARD dedicated symbols that are now
useless.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango committed:
soc: stm32l5: Rename overlays for nucleo_l552ze_q ns target.

File name should now be nucleo_l552ze_q_stm32l552xx_ns.overlay

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

nordicjm commented on boards/st/nucleo_l552ze_q/Kconfig.defconfig:
```diff
@@ -3,10 +3,7 @@
+if BOARD_NUCLEO_L552ZE_Q_STM32L552XX_NS
 
 if TRUSTED_EXECUTION_NONSECURE
 
```
aren't these 2 basically the same since you the ns part can only be non-secure?

nordicjm commented on boards/st/nucleo_l552ze_q/Kconfig.nucleo_l552ze_q:
```diff
@@ -0,0 +1,9 @@
+
+config BOARD_NUCLEO_L552ZE_Q_STM32L552XX_NS
+	bool
+	select SOC_STM32L552XX
```
```suggestion
```

nordicjm commented on boards/st/nucleo_l552ze_q/doc/nucleol552ze_q.rst:
```diff
@@ -177,8 +177,8 @@ The default configuration can be found in the defconfig and dts files:
 
-  - :zephyr_file:`boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns_defconfig`
-  - :zephyr_file:`boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_ns.dts`
+  - :zephyr_file:`boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_stm32l552xx_ns_defconfig`
+  - :zephyr_file:`boards/arm/nucleo_l552ze_q/nucleo_l552ze_q_stm32l552xx_ns.dts`
 
```
links are wrong

erwango replied:
Fixed, txs

nordicjm commented on boards/st/stm32l562e_dk/Kconfig.stm32l562e_dk:
```diff
@@ -0,0 +1,9 @@
+
+config BOARD_STM32L562E_DK_STM32L562XX_NS
+	bool
+	select SOC_STM32L562XX
```
```suggestion
```

nordicjm replied:
~~Not resolved~~ please amend commits in future rather than adding to the final commit with an incorrect commit

nordicjm commented on boards/st/nucleo_l552ze_q/Kconfig.nucleo_l552ze_q:
```diff
@@ -0,0 +1,9 @@
+
+config BOARD_NUCLEO_L552ZE_Q_STM32L552XX_NS
+	bool
+	select SOC_STM32L552XX
```
~~Not resolved~~ please amend commits in future rather than adding to the final commit with an incorrect commit

erwango replied:
`an incorrect commit`. Did you read commit description ?
 This is actually a dedicated refactoring commit with a proper justification.

nordicjm replied:
>  soc: stm32l5: Rename overlays for nucleo_l552ze_q ns target.
> 
> File name should now be nucleo_l552ze_q_stm32l552xx_ns.overlay

No relation to this change at all

nordicjm commented on boards/st/nucleo_l552ze_q/board.yml:
```diff
@@ -0,0 +1,7 @@
+  name: nucleo_l552ze_q
+  vendor: st
+  socs:
+    - name: stm32l552xx
```
- name should go left by 2 spaced, then all lines below also

erwango replied:
This is the way it is generated by `scripts/utils/board_v1_to_v2.py` script.

nordicjm replied:
@gmarull

gmarull replied:
our `.yamllint` file says `indentation: spaces: 2`, I guess it would complain if this was wrong.

gmarull replied:
@fabiobaltieri ?

fabiobaltieri replied:
from https://github.com/zephyrproject-rtos/zephyr/blob/main/.yamllint

```
  indentation:
    spaces: 2
    indent-sequences: consistent
```

the lists are a bit odd for some reasons, the linter just checks that the stile is coherent throughout the file

nordicjm commented on boards/st/stm32l562e_dk/board.yml:
```diff
@@ -0,0 +1,7 @@
+  name: stm32l562e_dk
+  vendor: st
+  socs:
+    - name: stm32l562xx
```
And here

nordicjm commented:
CI is failing for `nucleo_f722ze` and another stm32 board, not part of this conversion but were converted previously

erwango commented:
> CI is failing for `nucleo_f722ze` and another stm32 board, not part of this conversion but were converted previously

It is actually failing on main branch as well. We'll fix it there.
